ArduPlane/quadplane.cpp

```diff
@@ -1410,7 +1410,16 @@ void SLT_Transition::update()
         (quadplane.transition_failure > 0) &&
         ((now - transition_start_ms) > ((uint32_t)quadplane.transition_failure * 1000))) {
         gcs().send_text(MAV_SEVERITY_CRITICAL, "Transition failed, exceeded time limit");
-        plane.set_mode(plane.mode_qland, ModeReason::VTOL_FAILED_TRANSITION);
+        // for STL if option is set and ground speed> 1/2 arspd_fbw_min, then complete transition, otherwise QLAND.
+        // all others will immediately transition
+        if (quadplane.options & QuadPlane::OPTION_TRANS_FAIL_TO_FW) {
+            if (plane.quadplane.tailsitter.enable == 0 && plane.quadplane.tiltrotor.enable == 0
```

Comment:
```suggestion
            if (!plane.quadplane.tiltrotor.enabled()
```

Tailsitters don't use this code anymore, tiltrotors still do, Its better to check the `enabled` method rather than the enable value.

Reply:
the logic is to check for STL, have to include that its not tilt rotot and tailsitter in that logic...did not find a quadplane.is_stl() function

Reply:
Since the transition split tailsitters do not run this code at all. The function is in the class `SLT_Transition` for tailsitters we call this update in the `Tailsitter_Transition` class:

https://github.com/ArduPilot/ardupilot/blob/75959450ca27bbea4a6ea9e057ba8490b5735d49/ArduPlane/tailsitter.cpp#L673-L717

You should not find any checks of tailsitter enable in the transition code anymore, although as you say we still need them in the rest of the quadplane code.

Reply:
called the bool method now instead of the variable....